app/src/main/java/com/example/giphyapplication/ui/features/gifdetails/GifDetailsScreen.kt

```diff
@@ -3,85 +3,108 @@ package com.example.giphyapplication.ui.features.gifdetails
+import com.example.giphyapplication.composable.widgets.TopBar
+import com.example.giphyapplication.domain.model.Gif
+import com.example.giphyapplication.ui.features.gifdetails.contract.GifDetailsUiState
 
-@ExperimentalMaterialApi
-@Composable
-private fun ScreenConfig(
-    onIntent: (GifDetailsIntent) -> Unit,
-    content: @Composable () -> Unit,
-) = BaseScreen(
-    padding = PaddingValues(0.dp),
-    errorStateConfig = ErrorStateConfig(),
-    withBackButton = true,
-    onNavigateUp = {
-        onIntent(GifDetailsIntent.OnNavigateUp)
-    },
-    content = content
-)
-
-@ExperimentalMaterialApi
 @Composable
-fun GifDetailsRoute(
-    gif: Gif,
+fun GifDetailsScreen(
+    id: String,
     viewModel: GifDetailsViewModel,
-    navController: NavHostController,
+    onNavigateUp: () -> Unit,
 ) {
     val uiState by viewModel.uiState.collectAsStateWithLifecycle()
 
-    LaunchedEffect(key1 = viewModel) {
-        viewModel.acceptIntent(GifDetailsIntent.SetGifDetails(gif))
-    }
-
-    HandleEvents(
-        viewModel.event,
-        navController,
-    )
-
-    GifDetailsScreen(
+    GifDetailsScreenLoader(
+        id = id,
         uiState = uiState,
         onIntent = viewModel::acceptIntent,
+        onNavigateUp = onNavigateUp,
     )
-
 }
 
-@ExperimentalMaterialApi
 @Composable
-fun GifDetailsScreen(
+private fun GifDetailsScreenLoader(
+    id: String,
     uiState: GifDetailsUiState,
     onIntent: (GifDetailsIntent) -> Unit,
-) = ScreenConfig(onIntent) {
+    onNavigateUp: () -> Unit,
+) {
+    LaunchedEffect(key1 = 1) {
+        onIntent(GifDetailsIntent.GetGifDetails(id))
+    }
 
-    val configuration = LocalConfiguration.current
-    val size = configuration.screenWidthDp - 32
+    Scaffold(
+        topBar = {
+            TopBar(
+                withBackButton = true,
+                onNavigateUp = onNavigateUp,
+            )
+        },
+        content = { innerPadding ->
+            MainContent(
+                modifier = Modifier
+                    .padding(innerPadding),
+                viewState = uiState.viewState,
+                gif = uiState.gif
+            )
+        }
+    )
+}
+
+@Composable
+private fun MainContent(
+    modifier: Modifier,
+    viewState: GifDetailsViewState,
+    gif: Gif?
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .padding(dimensionResource(id = R.dimen.margin_normal)),
+    ) {
+        when (viewState) {
+            is GifDetailsViewState.Loading -> {
+                LoadingStateView()
+            }
+            is GifDetailsViewState.Error -> {
+                ErrorStateView(onAction = {})
+            }
+            is GifDetailsViewState.Success -> {
+                gif?.let {
+                    GifDetails(it)
+                }
+            }
+        }
+    }
+}
 
+@Composable
+private fun GifDetails(
+    gif: Gif
+) {
     val imageLoader = ImageLoader.Builder(LocalContext.current)
```

Comment:
Can we move this to a class of function?
NOTE: you're instantiating a class inside a composable function so you should use `remember` for this logic.